src/routes/api/users.js

```diff
@@ -1,5 +1,14 @@
 import { Router } from 'express';
+import { User } from '../../controllers';
+import { UserMiddleware } from '../../middlewares';
 
 const router = Router();
 
+const { userProfile, updateProfile } = User;
+const { isAuthenticated } = UserMiddleware;
+
+router.get('/profile/:userId', isAuthenticated, userProfile);
+router.get('/profile/:userId/edit', isAuthenticated, userProfile);
+router.put('/profile/:userId/update', isAuthenticated, updateProfile);
+
 export default router;
```

Comment:
I have a suggestion about how we define our API routes. See this snippet from this [article](https://restfulapi.net/resource-naming/)
### Never use CRUD function names in URIs
URIs should not be used to indicate that a CRUD function is performed. URIs should be used to uniquely identify resources and not any action upon them. HTTP request methods should be used to indicate which CRUD function is performed.

HTTP GET http://api.example.com/device-management/managed-devices  //Get all devices
HTTP POST http://api.example.com/device-management/managed-devices  //Create new Device

HTTP GET http://api.example.com/device-management/managed-devices/{id}  //Get device for given Id
HTTP PUT http://api.example.com/device-management/managed-devices/{id}  //Update device for given Id
HTTP DELETE http://api.example.com/device-management/managed-devices/{id}  //Delete device for given Id

Going by this standard, there shouldn't be a `router.get('/profile/:userId/edit', isAuthenticated, userProfile);` route. I understand the purpose of that route is to call it whenever you want to display a profile edit form. If it was to be a monolith, then it'd be fine but since it's an API, you can call this route `router.get('/profile/:userId', isAuthenticated, userProfile);` when ever you want to display the edit form and this `router.put('/profile/:userId/update', isAuthenticated, updateProfile);` (without the /update part) when the form is submitted.